src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -303,16 +325,26 @@ public Secret getPassword() {
     return this.getToken(null).getToken();
   }
 
+  /**
+   * Returns the Password.
+   *
+   * @param owner, owner of the repo.
+   * @return the password
+   */
+  public Secret getPassword(final String owner) {
+    return this.getToken(null, owner).getToken();
+  }
+
   /** {@inheritDoc} */
   @NonNull
   @Override
   public String getUsername() {
     return appID;
   }
 
-  private AppInstallationToken getCachedToken() {
+  private List<AppInstallationToken> getCachedTokens() {
     synchronized (this) {
-      return cachedToken;
+      return cachedTokens;
```

Comment:
Similar to other recent changes, we should look at using concurrent hash map instead.